lib/matplotlib/legend_handler.py

```diff
@@ -231,7 +231,7 @@ def create_artists(self, legend, orig_handle,
         xdata, xdata_marker = self.get_xdata(legend, xdescent, ydescent,
                                              width, height, fontsize)
 
-        ydata = ((height - ydescent) / 2.) * np.ones(xdata.shape, float)
+        ydata = np.full(xdata.shape, ((height - ydescent) / 2.), float)
```

Comment:
may as well drop the dot after "/2" while you're at it, the `__future__`(division) is now